OpenRA.Mods.RA2/Traits/Garrisoner.cs

```diff
@@ -0,0 +1,190 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Mods.RA2.Activities;
+using OpenRA.Mods.RA2.Orders;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA2.Traits
+{
+    public enum AlternateGarrisonMode { None, Force, Default, Always }
+
+    [Desc("This actor can enter Garrisonable actors.")]
+    public class GarrisonerInfo : ITraitInfo
+    {
+        public readonly string GarrisonType = null;
+        public readonly PipType PipType = PipType.Green;
+        public readonly int Weight = 1;
+
+        [Desc("Use to set when to use alternate transports (Never, Force, Default, Always).",
+            "Force - use force move modifier (Alt) to enable.",
+            "Default - use force move modifier (Alt) to disable.")]
+        public readonly AlternateGarrisonMode AlternateGarrisoningMode = AlternateGarrisonMode.Default;
+
+        [Desc("Number of retries using alternate transports.")]
+        public readonly int MaxAlternateTransportAttempts = 1;
+
+        [Desc("Range from self for looking for an alternate transport (default: 5.5 cells).")]
+        public readonly WDist AlternateTransportScanRange = WDist.FromCells(11) / 2;
+
+        [Desc("Valid Stances. Default Ally")]
+        public readonly Stance TargetStances = Stance.Ally;
```

Comment:
Shouldn't this also have Neutral to assume the RA2 defaults?

Reply:
I am not sure what this is even here for. My code is a refactored from DnApp's original PR:
https://github.com/OpenRA/OpenRA/pull/12774/files#diff-bf5850519488dffb5bcbdd86f62009eaR42
he added this to passenger. And I just took a lot of what he had as is but encapsulated the functionality into their own traits (per a lot of the comments in the PR, it looked like this was very much asked for).

The only place this Property is referenced is here: https://github.com/MustaphaTR/Romanovs-Vengeance/pull/13/files#diff-24d7408fd3ae598f11a5598df88c406cR98


In the original PR you questioned DnApp on it's use / wording. Perhaps you could help me to understand what we are looking for here. 

I guess I'll start by checking out what this function actually does in the debugger.

Reply:
See my other comment about what this is actually doing. Since neutrals own the garrisonable buildings on the map, having both Ally and Neutral listed for start would be beneficial at this point, I guess.